extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/attribute/RequestLineAttribute.java

```diff
@@ -23,8 +23,20 @@ public String readAttribute(final RoutingContext exchange) {
                 .append(exchange.request().method())
                 .append(' ')
                 .append(exchange.request().uri());
-        sb.append(' ')
-                .append(exchange.request().version());
+        sb.append(' ');
+        String httpVersion = "-";
+        switch (exchange.request().version()) {
+            case HTTP_1_0:
+                httpVersion = "HTTP/1.0";
+                break;
+            case HTTP_1_1:
+                httpVersion = "HTTP/1.1";
+                break;
+            case HTTP_2:
+                httpVersion = "HTTP/2";
+                break;
+        }
```

Comment:
Wondering if a more future proof option would be to transform the enum name into the version.

Something like enumName.replace("HTTP_", "HTTP/").replace("_", "."). WDYT?

Reply:
Hello @gsmet,
I've updated the PR to include this suggestion. It's a best effort code to try and infer the HTTP version from the enum value. Essentially we are relying on whoever adds a new enum constant to follow the existing name convention in that enum. I didn't add too much and too complicated logic here since if this does end up reporting a "wrong" value in the log file, fixing it will be a simple change (to add the new case statement).